posthog/tasks/detect_alerts_anomalies.py

```diff
@@ -0,0 +1,77 @@
+from typing import cast
+
+import structlog
+from celery import shared_task
+from django.utils import timezone
+
+from posthog.email import EmailMessage
+from posthog.hogql_queries.legacy_compatibility.filter_to_query import filter_to_query
+from posthog.hogql_queries.query_runner import get_query_runner
+from posthog.models import Alert, AnomalyCondition
+from posthog.schema import HogQLQueryResponse
+
+logger = structlog.get_logger(__name__)
+
+
+def check_all_alerts() -> None:
+    alerts = Alert.objects.all().only("id")
+    for alert in alerts:
+        logger.info("scheduling alert", alert_id=alert.id)
+        check_alert.delay(alert.id)
+
+
+@shared_task(ignore_result=True)
+def check_alert(id: int) -> None:
+    alert = Alert.objects.get(pk=id)
+    insight = alert.insight
+    if not insight.query:
```

Comment:
We should defer any query logic to existing methods, basically how it's done here

https://github.com/PostHog/posthog/blob/dd4be4fc3f97a85660db3e068c4ef002ddd5976b/posthog/api/insight.py#L575-L594

Execution mode should be `ExecutionMode.RECENT_CACHE_CALCULATE_BLOCKING_IF_STALE` so we leverage cache if available (and we have the benefit for the user of updating it here).

Reply:
Yeah, thanks, I was looking for something like this